_posts/2015-08-13-Next-year.md

```diff
@@ -66,13 +66,71 @@ Let's look at some of the biggest plans in each of these categories.
 
 #### Doubling down: infrastructure investments
 
-**Crater**.
-
-**Incremental compilation**. [MIR][mir]
+**Crater**. One of our big goals with Rust is that upgrades between
+versions should be "hassle-free". An important part of this story is
+detecting compiler bugs that cause code to stop working. Naturally,
+the compiler has its own large test suite, but that is only a small
+fraction of the code that's out there "in the wild". [Crater] is a
+tool that aims to close that gap by testing the compiler against all
+the packages found in [crates.io], giving us a much better idea
+whether any code has stopped compiling recently.
+
+We've been using the current version of crater for some time. For
+example, we regularly compare the nightly release against the latest
+stable build. We also use crater to check in-progress branches and
+estimate the impact of a change. 
+
+Interestingly, we have often found that when code stops compiling,
+it's not because of a bug in the compiler. Rather, it's because we
+*fixed* a bug, and that code happened to be relying on the older
+behavior. Even in those cases, using crater lets us improve the
+experience. For example, for changes that affect a lot of projects,
+we've often chosen to phase in the change gradually. In such cases, we
+issue one release where users get warnings, and only start issuing
+errors on the next release, giving people time to migrate their code.
+
+Over the next year or so, we plan to improve crater in numerous ways.
+For example, we'd like to extend the coverage to other platforms (not
```

Comment:
Should mention: getting it to run test suites. Also, just getting more coverage where there are more "interesting" dependencies.